From adb6a8043cac315730c231da5e70cb80570a9b7b Mon Sep 17 00:00:00 2001
From: Florian Boucault <florian.boucault@canonical.com>
Date: Tue, 26 Apr 2011 17:50:54 +0200
Subject: [PATCH] Immediatly delete obj in release() rather than using deleteLater()

Fixes a memory leak in unity-2d-places, see:
https://bugs.launchpad.net/unity-2d/+bug/723956

Test-case to reproduce:
---
import Qt 4.7

Item {
    Repeater {
        id: repeater
        delegate: Item{
                  }
    }
    focus: true
    Keys.onPressed: {
        if (event.key == Qt.Key_Return) {
            var i
            for (i=0; i<300; i++) {
                repeater.model = 100
                repeater.model = 0
            }
        }
    }
}
---
---
 .../graphicsitems/qdeclarativevisualitemmodel.cpp  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
index 065a388..a7884c0 100644
--- a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
+++ b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
@@ -994,7 +994,7 @@ QDeclarativeVisualDataModel::ReleaseFlags QDeclarativeVisualDataModel::release(Q
                 item->scene()->removeItem(item);
         }
         stat |= Destroyed;
-        obj->deleteLater();
+        delete obj;
     } else if (!inPackage) {
         stat |= Referenced;
     }
-- 
1.7.4.1
